OpenOversight/tests/routes/test_officer_and_department.py

```diff
@@ -103,6 +104,30 @@ def test_user_can_access_officer_list(mockdata, client, session):
         assert "Officers" in rv.data.decode("utf-8")
 
 
+@pytest.mark.parametrize(
+    "filter_func, has_placeholder",
+    [
+        (Operators.__invert__, True),
+        (lambda x: x, False),
+    ],
+)
+def test_officer_appropriately_shows_placeholder(
+    filter_func, has_placeholder, mockdata, client, session
+):
+    with current_app.test_request_context():
+        officer = Officer.query.filter(filter_func(Officer.face.any())).first()
+        placeholder = url_for(
+            "static", filename="images/placeholder.png", _external=True
+        )
+
+        rv = client.get(
+            url_for("main.officer_profile", officer_id=officer.id),
+            follow_redirects=True,
+        )
+
+        assert (placeholder not in rv.data.decode("utf-8")) ^ has_placeholder
```

Comment:
Why use the carat operator instead of `==`? I'm kind of confused by this assertion actually... Could it just be `placeholder in rv.data.decode('utf-8') == has_placeholder`? The `not in` is throwing me for a loop.

Reply:
OH that's a great idea, I've used the XOR operator (`^`) in the past for similar cases but this one is a bit more complex. I like your approach more!